### PR TITLE
Add cstruct 2.4.1

### PR DESCRIPTION
**Pull request by samoht**

### CODE REVIEW

**Comment by avsm** [comment]
datakit-client failing in revdeps:

```
#=== ERROR while compiling datakit-client.0.9.0 ===============================#
# command              ocaml pkg/pkg.ml build --pinned false -n datakit-client
# path                 /home/opam/.opam/ocaml-base-compiler.4.03.0/.opam-switch/build/datakit-client.0.9.0
# exit-code            1
# env-file             /home/opam/.opam/log/datakit-client-67-8000ab.env
# output-file          /home/opam/.opam/log/datakit-client-67-8000ab.out
### output ###
# ocamlfind ocamlopt unix.cmxa -I /home/opam/.opam/ocaml-base-compiler.4.03.0/lib/ocamlbuild /home/opam/.opam/ocaml-base-compiler.4.03.0/lib/ocamlbuild/ocamlbuildlib.cmxa -linkpkg myocamlbuild.ml /home/opam/.opam/ocaml-base-compiler.4.03.0/lib/ocamlbuild/ocamlbuild.cmx -o myocamlbuild
# ocamlfind ocamldep -package 'protocol-9p uri' -package 'bytes lwt astring logs result cstruct fmt rresult' -modules src/datakit-client/datakit_client_9p.ml > src/datakit-client/datakit_client_9p.ml.depends
# ocamlfind ocamldep -package 'protocol-9p uri' -package 'bytes lwt astring logs result cstruct fmt rresult' -modules src/datakit-client/datakit_client_9p.mli > src/datakit-client/datakit_client_9p.mli.depends
# ocamlfind ocamldep -package 'protocol-9p uri' -package 'bytes lwt astring logs result cstruct fmt rresult' -modules src/datakit-client/datakit_S.mli > src/datakit-client/datakit_S.mli.depends
# ocamlfind ocamldep -package 'protocol-9p uri' -package 'bytes lwt astring logs result cstruct fmt rresult' -modules src/datakit-client/datakit_path.mli > src/datakit-client/datakit_path.mli.depends
# ocamlfind ocamlc -c -g -bin-annot -safe-string -package 'protocol-9p uri' -package 'bytes lwt astring logs result cstruct fmt rresult' -warn-error +1..49-3 -w A-4-41-44 -I src/datakit-client -I src -o src/datakit-client/datakit_path.cmi src/datakit-client/datakit_path.mli
# ocamlfind ocamlc -c -g -bin-annot -safe-string -package 'protocol-9p uri' -package 'bytes lwt astring logs result cstruct fmt rresult' -warn-error +1..49-3 -w A-4-41-44 -I src/datakit-client -I src -o src/datakit-client/datakit_S.cmi src/datakit-client/datakit_S.mli
# ocamlfind ocamlc -c -g -bin-annot -safe-string -package 'protocol-9p uri' -package 'bytes lwt astring logs result cstruct fmt rresult' -warn-error +1..49-3 -w A-4-41-44 -I src/datakit-client -I src -o src/datakit-client/datakit_client_9p.cmi src/datakit-client/datakit_client_9p.mli
# + ocamlfind ocamlc -c -g -bin-annot -safe-string -package 'protocol-9p uri' -package 'bytes lwt astring logs result cstruct fmt rresult' -warn-error +1..49-3 -w A-4-41-44 -I src/datakit-client -I src -o src/datakit-client/datakit_client_9p.cmi src/datakit-client/datakit_client_9p.mli
# File "src/datakit-client/datakit_client_9p.mli", line 3, characters 18-38:
# Error: Unbound module Protocol_9p_client
# Command exited with code 2.
# pkg.ml: [ERROR] cmd ['ocamlbuild' '-use-ocamlfind' '-classic-display' '-tag' 'debug'
#      '-build-dir' '_build' '-Is' 'src,src/datakit-client' 'CHANGES.md'
#      'LICENSE.md' 'README.md' 'pkg/META.client' 'datakit-client.opam'
#      'src/datakit-client/datakit-client.a'
#      'src/datakit-client/datakit-client.cmxs'
#      'src/datakit-client/datakit-client.cmxa'
#      'src/datakit-client/datakit-client.cma'
#      'src/datakit-client/datakit_path.cmx'
#      'src/datakit-client/datakit_path.cmi'
#      'src/datakit-client/datakit_path.mli'
#      'src/datakit-client/datakit_client_9p.cmx'
#      'src/datakit-client/datakit_client_9p.cmi'
#      'src/datakit-client/datakit_client_9p.mli'
#      'src/datakit-client/datakit_S.mli' 'src/datakit-client/datakit_S.cmi'
#      'src/datakit-client/mount.native']: exited with 10

```

**Comment by avsm** [comment]
xe revdep failure fixed in #9082 -- rest is green!

**Comment by samoht** [comment]
@avsm all datakit-* revdeps are being fixed by https://github.com/ocaml/opam-repository/pull/9076

File: packages/cstruct/cstruct.2.4.1/opam [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  [make "js-uninstall"]
+  ["ocamlfind" "remove" "cstruct"]
+]
+depends: [
+  "ocamlfind" {build}
+  "ounit"     {test}
+  "ocplib-endian"
+  "sexplib"
+  "base-bytes"
+  "ocaml-migrate-parsetree"
```
Comment (samoht) [inline packages/cstruct/cstruct.2.4.1/opam]: @avsm @let-def does it needs a version constraint to be compatible with https://github.com/ocaml/opam-repository/pull/9029 (not sure why that other PR requires this one to be merged first)

**Pull request merged**